prism/src/main/java/org/apache/falcon/resource/channel/HTTPChannel.java

```diff
@@ -170,8 +171,20 @@ private String pathValue(Method method, Object... args) throws FalconException {
                     String annotationClass = paramAnnotation.annotationType().getName();
 
                     if (annotationClass.equals(QueryParam.class.getName())) {
-                        queryString.append(getAnnotationValue(paramAnnotation, "value")).
-                                append('=').append(arg).append("&");
+                        if (args[index] instanceof List){
+                            List lifecycle = (List) args[index];
```

Comment:
It should be valid for all list query params. why did you specify lifecycle here ?